app/supplier_constants.py

```diff
@@ -0,0 +1,15 @@
+# Here we define a set of hardcoded keys that we use when denormalizing data from Supplier/ContactInformation tables
```

Comment:
I'd be tempted to get rid of this :man_shrugging: 

It's not a pattern we're using elsewhere.
We only us them in the application code once (In the SQL)
They should probably be hard coded in the tests so that people reading the tests can see exactly what the result looks like without having to chase the variables about. Verbose tests are not a bad thing :man_shrugging:

Reply:
It's a good point about the tests, however I think keeping the hardcoded keys out of the SQL statement is reasonable. (If we do keep this as a separate file remember to update the imports elsewhere!)

Reply:
Yeah my reasoning is as Kat says here. I think it's good to keep them separate from the statement itself. I also don't think it's that valuable to require devs to update the test code when we change the keys in the core application code. The value of the variable doesn't really matter - devs just need to know that it's a "magic" string we're using as names.

Reply:
I think it's good to have these "magic keys" clearly listed outside of the SQL, even if they are only used here in this one place at the moment.  

As to whether or not we should be using them in tests... I'm kind-of undecided.  There's always an element of marking your own homework if tests refer back to things in the main codebase, and best practice says don't do it.  But I also see Sam's point about the values not actually mattering in terms of code correctness.  The only thing you're going to detect is a typo, but maybe that's still useful... If we end up with a load of declarations with a `suplpierRegisterdCounty`instead of `supplierRegisteredCountry` it could matter.